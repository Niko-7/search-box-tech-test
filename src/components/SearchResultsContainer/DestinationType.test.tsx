/*eslint-disable */
import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { DestinationType } from ".";
const { axe, toHaveNoViolations } = require("jest-axe");
expect.extend(toHaveNoViolations);

afterEach(() => {
  cleanup();
});

describe("DestinationType component", () => {
  describe("It should return the appropriate type of destination", () => {
    it("should return City", () => {
      render(<DestinationType destinationKind={"city-123"} />);
      const typeCity = screen.getByText("City");
      expect(typeCity).toBeInTheDocument;
    });
    it("should return Airport", () => {
      render(<DestinationType destinationKind={"airport-123"} />);
      const typeAirport = screen.getByText("Airport");
      expect(typeAirport).toBeInTheDocument;
    });
    it("should return Station", () => {
      render(<DestinationType destinationKind={"station-123"} />);
      const typeStation = screen.getByText("Station");
      expect(typeStation).toBeInTheDocument;
    });
  });
});

describe("Snapshot tests", () => {
  it("should match the snapshot of the DestinationType component", () => {
    const tree = renderer
      .create(<DestinationType destinationKind={"city-123"} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("accesibility tests", () => {
  it("should have no accesibility errors on DestinationType component", async () => {
    const { container } = render(
      <DestinationType destinationKind={"city-123"} />
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
