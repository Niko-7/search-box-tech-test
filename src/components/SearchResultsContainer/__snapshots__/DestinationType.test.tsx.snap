// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests should match the snapshot of the DestinationType component 1`] = `
<div
  className="search-results__location-type "
>
  <span>
    
  </span>
</div>
`;
